apollo-router-core/src/spec/schema.rs

```diff
@@ -236,17 +237,63 @@ impl std::str::FromStr for Schema {
             })
             .collect();
 
-        let fragments = Fragments::from(&document);
+        let enums: HashMap<String, HashSet<String>> = document
+            .definitions()
+            .filter_map(|definition| match definition {
+                // Spec: https://spec.graphql.org/draft/#sec-Enums
+                ast::Definition::EnumTypeDefinition(definition) => {
+                    let name = definition
+                        .name()
+                        .expect("the node Name is not optional in the spec; qed")
```

Comment:
Wider question: If `enum_values_definition` is not optional in the spec why is it optional in the AST? Are we risking panics due to a malformed schema?

Reply:
everything in the AST is under options (Ex: https://docs.rs/apollo-parser/latest/apollo_parser/ast/struct.EnumValuesDefinition.html ) because it is not yet a representation that is completely validated. The AST can be used to represent an invalid document that was just parsed or is currently edited.
That's why we need the semantic analysis part.
At this point of the code unwraps should be safe though, since the query and schema were already validated by the query planner

Reply:
SGTM